src/sdl2_extras.c

```diff
@@ -0,0 +1,198 @@
+	return base ? base : "";
+}
+
+static const char *get_pref_path(const char *org, const char *app)
+{
+	const char *pref = NULL;
+
+	if (org != NULL && app != NULL) {
+		pref = SDL_GetPrefPath(org, app);
+	}
+
+	return pref ? pref : "";
+}
+
+int spnlib_SDL_GetPaths(SpnValue *ret, int argc, SpnValue *argv, void *ctx)
+{
+	if (argc >= 2) {
+		CHECK_ARG_RETURN_ON_ERROR(0, string);
+		CHECK_ARG_RETURN_ON_ERROR(1, string);
+	}
+
+	// construct return value: hashmap with strings
+	*ret = spn_makehashmap();
+	SpnHashMap *paths = spn_hashmapvalue(ret);
+
+	// get proper arguments for Pref, if given
+	const char *org = NULL, *app = NULL;
+	if (argc >= 2) {
+		org = STRARG(0);
+		app = STRARG(1);
+	}
+
+	SpnValue base = spn_makestring_nocopy(get_base_path());
+	SpnValue pref = spn_makestring_nocopy(get_pref_path(org, app));
+
+	spn_hashmap_set_strkey(paths, "base", &base);
+	spn_hashmap_set_strkey(paths, "pref", &pref);
+
+	spn_value_release(&base);
+	spn_value_release(&pref);
+
+	return 0;
+}
+
+/////////////////////////////////
+//////    SDL's version    //////
+/////////////////////////////////
+int spnlib_SDL_GetVersion(SpnValue *ret, int argc, SpnValue *argv, void *ctx)
+{
+	*ret = spn_makehashmap();
+	SpnHashMap *version = spn_hashmapvalue(ret);
+
+	SDL_version linked;
+	SDL_GetVersion(&linked);
+
+	SpnValue major = spn_makeint(linked.major);
+	SpnValue minor = spn_makeint(linked.minor);
+	SpnValue patch = spn_makeint(linked.patch);
+
+	spn_hashmap_set_strkey(version, "major", &major);
+	spn_hashmap_set_strkey(version, "minor", &minor);
+	spn_hashmap_set_strkey(version, "patch", &patch);
+
+	spn_value_release(&major);
+	spn_value_release(&minor);
+	spn_value_release(&patch);
+
+	return 0;
+}
+
+/////////////////////////////////
+////    Platform and CPU    /////
+/////////////////////////////////
+int spnlib_SDL_GetPlatform(SpnValue *ret, int argc, SpnValue *argv, void *ctx)
+{
+	// Simplest function to bind
+	*ret = spn_makestring_nocopy(SDL_GetPlatform());
+
+	return 0;
+}
+
+int spnlib_SDL_GetCPUSpecs(SpnValue *ret, int argc, SpnValue *argv, void *ctx)
+{
+	*ret = spn_makehashmap();
+	SpnHashMap *cpu_features = spn_hashmapvalue(ret);
+
+	// These give integers
+	SpnValue cache = spn_makeint(SDL_GetCPUCacheLineSize());
+	SpnValue cores = spn_makeint(SDL_GetCPUCount());
+	SpnValue ram = spn_makeint(SDL_GetSystemRAM());
+	// And these give booleans
+	SpnValue now3d = SPN_SDLBOOL(SDL_Has3DNow());
+	SpnValue avx = SPN_SDLBOOL(SDL_HasAVX());
+	SpnValue avx2 = SPN_SDLBOOL(SDL_HasAVX2());
+	SpnValue altivec = SPN_SDLBOOL(SDL_HasAltiVec());
+	SpnValue mmx = SPN_SDLBOOL(SDL_HasMMX());
+	SpnValue rdtsc = SPN_SDLBOOL(SDL_HasRDTSC());
+	SpnValue sse = SPN_SDLBOOL(SDL_HasSSE());
+	SpnValue sse2 = SPN_SDLBOOL(SDL_HasSSE2());
+	SpnValue sse3 = SPN_SDLBOOL(SDL_HasSSE3());
+	SpnValue sse41 = SPN_SDLBOOL(SDL_HasSSE41());
+	SpnValue sse42 = SPN_SDLBOOL(SDL_HasSSE42());
+
+	// Assembling all this mess
+	spn_hashmap_set_strkey(cpu_features, "Cache (KB)", &cache);
```

Comment:
You are absolutely right. I was conflicted when I first wrote this, so I'll remove those details (as well as lowercase everything and explicitly mention these kB/MB details in the documents)